src/main/java/io/specto/hoverfly/junit/core/model/RequestMatcher.java

```diff
@@ -13,74 +13,150 @@
 
 import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.annotation.JsonInclude;
 import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
 import org.apache.commons.lang3.builder.EqualsBuilder;
 import org.apache.commons.lang3.builder.HashCodeBuilder;
+import org.apache.commons.lang3.builder.ToStringBuilder;
 
 import java.util.List;
 import java.util.Map;
 
+import static io.specto.hoverfly.junit.core.model.FieldMatcher.fromExactMatchString;
+
+@JsonIgnoreProperties(ignoreUnknown = true)
+@JsonInclude(JsonInclude.Include.NON_EMPTY)
 public class RequestMatcher {
-    private final String requestType;
-    private final String path;
-    private final String method;
-    private final String destination;
-    private final String scheme;
-    private final String query;
-    private final String body;
+
+    @JsonDeserialize(using = FieldMatcherDeserializer.class)
+    private final FieldMatcher path;
+    @JsonDeserialize(using = FieldMatcherDeserializer.class)
+    private final FieldMatcher method;
+    @JsonDeserialize(using = FieldMatcherDeserializer.class)
+    private final FieldMatcher destination;
+    @JsonDeserialize(using = FieldMatcherDeserializer.class)
+    private final FieldMatcher scheme;
+    @JsonDeserialize(using = FieldMatcherDeserializer.class)
+    private final FieldMatcher query;
+    @JsonDeserialize(using = FieldMatcherDeserializer.class)
+    private final FieldMatcher body;
     private final Map<String, List<String>> headers;
 
+    @Deprecated
+    public RequestMatcher(String path,
+                          String method,
+                          String destination,
+                          String scheme,
+                          String query,
+                          String body,
+                          Map<String, List<String>> headers) {
+        this.path = fromExactMatchString(path);
+        this.method = fromExactMatchString(method);
+        this.destination = fromExactMatchString(destination);
+        this.scheme = fromExactMatchString(scheme);
+        this.query = fromExactMatchString(query);
+        this.body = fromExactMatchString(body);
+        this.headers = headers;
+    }
+
+
     @JsonCreator
-    public RequestMatcher(@JsonProperty("path") String path,
-                          @JsonProperty("method") String method,
-                          @JsonProperty("destination") String destination,
-                          @JsonProperty("scheme") String scheme,
-                          @JsonProperty("query") String query,
-                          @JsonProperty("body") String body,
-                          @JsonProperty("headers") Map<String, List<String>> headers,
-                          @JsonProperty("requestType") String requestType) {
+    public RequestMatcher(@JsonProperty("path") FieldMatcher path,
```

Comment:
We could just double check the schema, and then have a `bodyMatcher`, `pathMatcher` etc that map to it correctly. I don't feel comfortable making it possible to create an entity which doesn't comply with the JSONSchema